### DISCUSSION TITLE
Updating CTk.ProgressBar in real time with reference to gathered info

### DISCUSSION THREAD

**Discussion by duruburak** [TomSchimansky/CustomTkinter#653] [Q&A]
I'm trying to make a progress bar that shows how much of the file downloaded at that moment in real time in terms of percentile.
I need to set the progress bar's length to 100 and I need something to update the value of the progress bar simultaneously with reference to the downloaded part of the file.
How can I do that?

**Comment by Akascape** [comment]
From where will you download?
Ig you are using pytube.

**Reply by duruburak** [reply]
pytube yes. I used this, it worked for me.
```
def on_progress(chunk, file_handle, bytes_remaining):
    global file_size
    file_size = chunk.filesize
    downloaded = file_size - bytes_remaining
    percent = downloaded/file_size    
    time.sleep(0.05)
    progress_label.set(percent)
    screen.update()
    
progress_label =  customtkinter.CTkProgressBar(frame_1, fg_color="white", width=380, height=50, corner_radius=100)
progress_label.pack(pady=50, padx=5)
```
